tasks/libs/common/go.py

```diff
@@ -16,7 +16,7 @@ def download_go_dependencies(ctx: Context, paths: list[str], verbose: bool = Fal
         for path in paths:
             with ctx.cd(path):
                 for retry in range(max_retry):
-                    result = ctx.run(f"go mod download{verbosity}")
+                    result = ctx.run(f"go mod download{verbosity}", warn=True)
                     if result.exited is None or result.exited > 0:
                         wait = 10 ** (retry + 1)
                         print(f"[{retry + 1} / {max_retry}] Failed downloading {path}, retrying in {wait} seconds")
```

Comment:
I believe we should explicitly fail if we use up all our retries - With the current code, we exit the loop without raising an error when it happens.

Reply:
Correct, fixed in https://github.com/DataDog/datadog-agent/pull/28258/commits/44033bec43c4a6db904704bfd3fa4feac4317e90. If we break (it means it worked) we don't raise the exception